torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp

```diff
@@ -369,13 +369,122 @@ size_t nRootDims(const TensorView* tv) {
   }
   return tv_n_dims;
 }
+
+// DomainMap uses the ComputeAtMap to find a reference TensorView
+// that maps to all iterDomains in the fusion.
+class DomainMap {
+ public:
+  DomainMap(Fusion* fusion)
+      : fusion_(fusion),
+        ca_index_map_(ComputeAtMap(ComputeAtMap::MappingMode::INDEX)) {
+    ca_index_map_.build(fusion);
```

Comment:
Are we creating the index map just here at the scheduling time? Building the map is relatively a heavyweight operation, so it should be reused if possible.